packages/extension-vscode/src/server.ts

```diff
@@ -0,0 +1,204 @@
+const loadUserConfig = (directory: string, Configuration: typeof config.Configuration): UserConfig => {
+    const defaultConfig: UserConfig = { extends: ['development'] };
+
+    try {
+        const configPath = Configuration.getFilenameForDirectory(directory);
+        const resolvedPath = path.resolve(directory, configPath);
+
+        return Configuration.loadConfigFile(resolvedPath) || defaultConfig;
+    } catch (e) {
+        return defaultConfig;
+    }
+};
+
+// Load a copy of webhint with the provided configuration
+const loadEngine = async (directory: string, configuration: config.Configuration): Promise<hint.Engine> => {
+    const { loadResources } = await loadModule<typeof loader>(directory, 'hint/dist/src/lib/utils/resource-loader');
+    const resources = loadResources(configuration);
+    const { Engine } = await loadModule<typeof hint>(directory, 'hint');
+
+    return new Engine(configuration, resources);
+};
+
+// Load both webhint and a configuration, adjusting it as needed for this extension
+const loadWebHint = async (directory: string): Promise<hint.Engine> => {
+    const { Configuration } = await loadModule<typeof config>(directory, 'hint/dist/src/lib/config');
+    const userConfig = loadUserConfig(directory, Configuration);
+
+    // The vscode extension only works with the local connector
+    userConfig.connector = { name: 'local' };
+
+    if (!userConfig.parsers) {
+        userConfig.parsers = [];
+    }
+
+    // Ensure the HTML parser is loaded
+    if (userConfig.parsers.indexOf('html') === -1) {
+        userConfig.parsers.push('html');
+    }
+
+    return await loadEngine(directory, Configuration.fromConfig(userConfig));
+};
+
+let workspace: string;
+let engine: hint.Engine;
+
+connection.onInitialize(async (params) => {
+    try {
+        // TODO: support multiple workspaces (`params.workspaceFolders`)
+        workspace = params.rootPath;
+        engine = await loadWebHint(workspace);
+    } catch (e) {
+        console.error(e);
+        connection.window.showErrorMessage('[webhint] Load failed. Add it via `npm install hint --save-dev`.');
+    }
+
+    return { capabilities: { textDocumentSync: documents.syncKind } };
+});
+
+// Translate a webhint severity into the VSCode DiagnosticSeverity format.
+const webhintToDiagnosticServerity = (severity: Severity): DiagnosticSeverity => {
+    switch (severity) {
+        case 2:
+            return DiagnosticSeverity.Error;
+        case 1:
+            return DiagnosticSeverity.Warning;
+        default:
+            return DiagnosticSeverity.Hint;
+    }
+};
+
+// Translate a webhint problem into the VSCode diagnostic format.
+const problemToDiagnostic = (problem: Problem): Diagnostic => {
+
+    let character = problem.location.column;
+    let line = problem.location.line;
+
+    // Move (-1, -1) to (0, 0) so VSCode underlines the start of the document.
+    if (character < 0 && line < 0) {
```

Comment:
I know we should never get something like `(-1, 10)`, but maybe we should just replace all `-1` with `0`?

Reply:
Works for me.

Reply:
@molant, how about I change this to `||`, as in `if (character < 0 || line < 0) {`. 

Even if we could get `(-1, 10)`, I think the best response would be to redirect to `(0, 0)` (which this would do).